apps/ledger-live-mobile/src/screens/VerifyAccount/index.tsx

```diff
@@ -49,7 +49,19 @@ export default function VerifyAccount({ navigation, route }: NavigationProps) {
   const onDone = useCallback(() => {
     const n = navigation.getParent<StackNavigatorNavigation<RootStackParamList>>();
 
-    if (n) {
+    // get the route at the given index on navigation
+    const { index, routes } = navigation.getState();
+    const { name: routeAtIndexName } = routes[index];
+
+    // if this route is a live app we do not want to pop
+    // as doing so would take the user out of the live app.
+    // an example of this is the BTCDirect app.
+    const isRouteAtIndexALiveApp = [ScreenName.Exchange, ScreenName.PlatformApp].includes(
```

Comment:
If you want get only live apps I think this filter could be incorrect. ScreenName.Exchange include sell and swap. 

![image](https://github.com/LedgerHQ/ledger-live/assets/38540290/2039a156-48b5-4d01-b0f4-233531fae5aa)

Reply:
Oh ok, I wasn't aware that it also included swap, so will have to double check that!

Yes, will update the comment also and try to better explain it.

I don't quite understand what you mean by this sentence 😄 
"On the other hand, do we need really filter "buy" app here?"

Reply:
Confirmed. After navigate to BTC Direct we continue in Buy context.